synapse/push/bulk_push_rule_evaluator.py

```diff
@@ -211,8 +214,15 @@ async def action_for_event_by_user(
             sender_power_level,
         ) = await self._get_power_levels_and_sender_level(event, context)
 
+        relations = await self.store.get_mutual_event_relations(event)
```

Comment:
I think we should only try to fetch this if any of the rules are of the `relation_match` kind, but maybe that's pre-mature optimization?

Reply:
If we're adding one to the default rules then I think it'd be pretty rare to not need it? Though maybe we want to only pull them out if the MSC is enabled?